### ISSUE TITLE
sqlalchemy on pyright 1.1.74

### ISSUE THREAD

**Issue by himat** [microsoft/pyright#1059]
Hi, what is the proper way to deal with the changes in pyright 1.1.74 when using sql alchemy?

```
 class Company(Base):
     __tablename__ = "company"
     id = Column(Integer, primary_key=True)
     name = Column(String)
```

pyright now throws an error when I try to do something like
```
new_company_name = "hi"
company = Company()
company.name = new_company_name
```
with the error that
```
Expression of type "str" cannot be assigned to member "name" of class "Company"
  "str" is incompatible with "Column[Text]" (reportGeneralTypeIssues)
```

I didn't get these errors on the previous versions, so what should I do?

**Comment by erictraut** [comment]
The problem is that the `name` variable has an inferred type of `Column`, but you are trying to assign a `str` to it. If the `Column` provided a `__set__` method, this would be permitted, but since it doesn't, the statement `company.name = new_company_name` is attempting to overwrite `name` with a `str`, which is not a compatible type.

If your intent is to allow `name` to contain either a `Column` object or a `str` object, then you should declare its type as such.

```python
    name: Union[Column, str] = Column(String)
```

**Comment by rmzr7** [comment]
Hi @erictraut is this still the preferred way to handle sqlalchemy columns? or is the preferred way now to declare name: `Column[str] = Column(String)`?

**Comment by erictraut** [comment]
I don't have any expertise with sqlalchemy, so I don't know the answer to that question. My answer above was a general statement about type checking rules. Perhaps other pyright users who have more experience with sqlalchemy could provide additional guidance here.
